torch/csrc/dynamo/guards.cpp

```diff
@@ -26,23 +26,20 @@ class TensorCheck {
       const LocalState& state,
       PyTypeObject* pt,
       const at::Tensor& v,
-      bool dynamic_shapes)
+      std::vector<std::optional<int64_t>> dynamic_dims)
       : pytype(pt),
         dispatch_key_(state.apply(v.key_set()).raw_repr()),
         dtype_(v.dtype().toScalarType()),
         device_index_(v.device().index()),
         requires_grad_(state.grad_mode_enabled && v.requires_grad()),
-        dynamic_shapes_(dynamic_shapes) {
+        sizes_(dynamic_dims) {
     dim_ = v.ndimension();
-    if (!dynamic_shapes_) {
-      const auto& sizes = v.sizes();
-      const auto& strides = v.strides();
-      sizes_.reserve(dim_);
-      strides_.reserve(dim_);
-      for (auto i : c10::irange(dim_)) {
-        sizes_.emplace_back(sizes[i]);
-        strides_.emplace_back(strides[i]);
-      }
+    const auto& strides = v.strides();
+    strides_.reserve(dim_);
+    for (auto i : c10::irange(dim_)) {
+      // Should we peek into dynamic_dims and determine if we need to make
+      // strides optionals?
+      strides_.emplace_back(strides[i]);
```

Comment:
You need both dynamic_sizes and dynamic_strides, right now you only transmit dynamic_sizes.